pinot-core/src/main/java/org/apache/pinot/core/transport/ServerChannels.java

```diff
@@ -139,22 +143,32 @@ private void attachSSLHandler(SocketChannel ch) {
       }
     }
 
-    synchronized void sendRequest(String rawTableName, AsyncQueryResponse asyncQueryResponse,
-        ServerRoutingInstance serverRoutingInstance, InstanceRequest instanceRequest)
+    private void sendRequestOrTimeOut(String rawTableName, AsyncQueryResponse asyncQueryResponse,
```

Comment:
Suggest renaming it to `sendRequest`. Passing in a timeout implies that it can time out

Reply:
done